tests/unit/Diff/Internal/TermListPatcherTest.php

```diff
@@ -0,0 +1,142 @@
+
+use Diff\DiffOp\Diff\Diff;
+use Diff\DiffOp\DiffOpAdd;
+use Diff\DiffOp\DiffOpChange;
+use Diff\DiffOp\DiffOpRemove;
+use PHPUnit_Framework_TestCase;
+use Wikibase\DataModel\Services\Diff\Internal\TermListPatcher;
+use Wikibase\DataModel\Term\Term;
+use Wikibase\DataModel\Term\TermList;
+
+/**
+ * @covers Wikibase\DataModel\Services\Diff\Internal\TermListPatcher
+ *
+ * @licence GNU GPL v2+
+ * @author Bene* < benestar.wikimedia@gmail.com >
+ */
+class TermListPatcherTest extends PHPUnit_Framework_TestCase {
+
+	public function providePatchTermList() {
+		return array(
+			'add a term' => array(
+				new TermList(),
+				new Diff( array(
+					'en' => new DiffOpAdd( 'foo' )
+				) ),
+				new TermList( array(
+					new Term( 'en', 'foo' )
+				) )
+			),
+			'change a term' => array(
+				new TermList( array(
+					new Term( 'en', 'foo' )
+				) ),
+				new Diff( array(
+					'en' => new DiffOpChange( 'foo', 'bar' )
+				) ),
+				new TermList( array(
+					new Term( 'en', 'bar' )
+				) )
+			),
+			'remove a term' => array(
+				new TermList( array(
+					new Term( 'en', 'foo' )
+				) ),
+				new Diff( array(
+					'en' => new DiffOpRemove( 'foo' )
+				) ),
+				new TermList()
+			),
+			'add an existing language is no-op' => array(
+				new TermList( array(
+					new Term( 'en', 'foo' )
+				) ),
+				new Diff( array(
+					'en' => new DiffOpAdd( 'bar' )
+				) ),
+				new TermList( array(
+					new Term( 'en', 'foo' )
+				) )
+			),
+			'add two terms' => array(
+				new TermList( array(
+					new Term( 'en', 'foo' )
+				) ),
+				new Diff( array(
+					'de' => new DiffOpAdd( 'bar' ),
+					'nl' => new DiffOpAdd( 'baz' )
+				) ),
+				new TermList( array(
+					new Term( 'en', 'foo' ),
+					new Term( 'de', 'bar' ),
+					new Term( 'nl', 'baz' )
+				) )
+			),
+			'remove a not existing language is no-op' => array(
+				new TermList( array(
+					new Term( 'en', 'foo' )
+				) ),
+				new Diff( array(
+					'de' => new DiffOpRemove( 'bar' )
+				) ),
+				new TermList( array(
+					new Term( 'en', 'foo' )
+				) )
+			),
+			'change a different value is no-op' => array(
+				new TermList( array(
+					new Term( 'en', 'foo' )
+				) ),
+				new Diff( array(
+					'en' => new DiffOpChange( 'bar', 'baz' )
+				) ),
+				new TermList( array(
+					new Term( 'en', 'foo' )
+				) )
+			),
+			'remove a different value is no-op' => array(
+				new TermList( array(
+					new Term( 'en', 'foo' )
+				) ),
+				new Diff( array(
+					'en' => new DiffOpRemove( 'bar' )
+				) ),
+				new TermList( array(
+					new Term( 'en', 'foo' )
+				) )
+			),
+			'complex diff' => array(
+				new TermList( array(
+					new Term( 'en', 'foo' ),
+					new Term( 'de', 'bar' ),
+					new Term( 'nl', 'baz' )
+				) ),
+				new Diff( array(
+					'en' => new DiffOpChange( 'foo', 'bar' ),
+					'de' => new DiffOpRemove( 'bar' ),
+					'nl' => new DiffOpChange( 'bar', 'foo' ),
+					'it' => new DiffOpAdd( 'foo' )
+				) ),
```

Comment:
When not stated explicitly, it is determined by seeing if there are changed that are no add or remove.